### PR TITLE
Improve textarea and fix bugs

### PR DESCRIPTION
**Pull request by PavelLaptev**
## ☕️ Reasoning
- Improve calculation method
- Fix PR textarea
- Remove old component
- update commit message with the new `textarea` component

### CODE REVIEW

File: packages/ui/src/lib/Textarea.svelte [hunk 1]
```diff
@@ -70,8 +73,24 @@
...
+
+			measureEl.style.fontFamily = textBoxElStyles.fontFamily;
+			measureEl.style.fontSize = textBoxElStyles.fontSize;
+			measureEl.style.fontWeight = textBoxElStyles.fontWeight;
+			measureEl.style.padding = textBoxElStyles.padding;
+			measureEl.style.width = textBoxElStyles.width;
+			measureEl.style.border = textBoxElStyles.border;
+			measureEl.style.width = textBoxElStyles.width;
+		}
+	});
```
Comment (Caleb-T-Owens) [inline packages/ui/src/lib/Textarea.svelte]: This ought to be an `$effect` block

File: packages/ui/src/lib/Textarea.svelte [hunk 2]
```diff
@@ -84,8 +103,10 @@
 		}
 	});
 
-	let maxHeight = fontSize * 1.5 * maxRows + padding.top + padding.bottom;
-	let minHeight = fontSize * 1.5 * minRows + padding.top + padding.bottom;
+	const lineHeight = 1.6;
+
+	let maxHeight = fontSize * maxRows + padding.top + padding.bottom;
+	let minHeight = fontSize * minRows + padding.top + padding.bottom;
```
Comment (Caleb-T-Owens) [inline packages/ui/src/lib/Textarea.svelte]: These depend on reactive properties (from `$props`) so these ought to be `$derived` statements

File: packages/ui/src/lib/Textarea.svelte (hunk 1 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/Textarea.svelte]: right!

**Review by Caleb-T-Owens — CHANGES_REQUESTED** [review]
A net loss of ~250 LOC, awesome!

It is however important that we fix these reactivity issues.

If you're dealing with **anything** that is declared in `$props` or with `$state`, you must then consume it via `$effect` or `$derived`, otherwise you will run into reactivity issues.

File: packages/ui/src/lib/Textarea.svelte (hunk 2 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/Textarea.svelte]: danke! Fixed

**Review by Caleb-T-Owens — APPROVED** [review]
Speedy Danke!

**Comment by PavelLaptev** [comment]
> A net loss of ~250 LOC, awesome!
> 
> It is however important that we fix these reactivity issues.
> 
> If you're dealing with **anything** that is declared in `$props` or with `$state`, you must then consume it via `$effect` or `$derived`, otherwise you will run into reactivity issues.

Thanks for pointing that out @Caleb-T-Owens

**Review by ndom91 — APPROVED** [review]

**Pull request merged**